launcher/jvm_tooling.cpp

```diff
@@ -166,30 +213,61 @@ JVM::JVM(const std::string &executable_path) {
   std::vector<std::string> jvm_args;
   if (!FLAGS_jvm_args.empty()) {
     jvm_args = splitEscaped(FLAGS_jvm_args);
+    for (const auto &arg : jvm_args) {
+      options.push_back(
+          JavaVMOption{.optionString = const_cast<char *>(arg.c_str())});
+    }
   }
-  for (const auto &arg : jvm_args) {
-    options.push_back(
-        JavaVMOption{.optionString = const_cast<char *>(arg.c_str())});
-  }
+
   std::vector<std::string> additional_jvm_args;
   if (!FLAGS_additional_jvm_args.empty()) {
     additional_jvm_args = splitEscaped(FLAGS_additional_jvm_args);
+    for (const auto &arg : additional_jvm_args) {
+      options.push_back(
+          JavaVMOption{.optionString = const_cast<char *>(arg.c_str())});
+    }
   }
-  for (const auto &arg : additional_jvm_args) {
-    options.push_back(
-        JavaVMOption{.optionString = const_cast<char *>(arg.c_str())});
+}
+
+JVM::JVM(const std::string &executable_path, bool android) {
+  JNI_CreateJavaVM_t CreateArtVM = LoadVMLibs(android);
+  std::string class_path = GetClassPath(executable_path, android);
+
+  std::vector<JavaVMOption> options;
+  options.push_back(
+      JavaVMOption{.optionString = const_cast<char *>(class_path.c_str())});
+  SetVMOptions(options, android);
+
+  jint jni_version = JNI_VERSION_1_8;
+  if (android) {
+    jni_version = JNI_VERSION_1_6;
```

Comment:
The why only up to 6 is supported, I dont know, but I know that Android only supports up to version 6 on JNI. 

There are people who know much more about this than me, but from my understanding we can build for Java 9 or 8, but after we desugar and dex we are going to have Java 7 with a subset of 8.

I added the java_language_version=9 because some flags I was using to desugar needed it, but I removed those. I will remove the java_language_version=9 and retest tomorrow to make sure that doesn't effect anything (it shouldn't) and push up that change.